src/vm/methodtablebuilder.cpp

```diff
@@ -1170,7 +1170,7 @@ BOOL MethodTableBuilder::CheckIfSIMDAndUpdateSize()
     STANDARD_VM_CONTRACT;
 
 #if defined(_TARGET_X86_) || defined(_TARGET_AMD64_)
-    if (!GetAssembly()->IsSIMDVectorAssembly())
+    if (!(GetAssembly()->IsSIMDVectorAssembly() || GetModule()->IsSystem()))
```

Comment:
This should check for intrinsic type. IsSystem is unnecessarily broad.

Reply:
intrinsic type isn't set yet at this point.

`CheckIfSIMDAndUpdateSize()` gets called earlier in `BuildMethodTableThrowing` here:
https://github.com/dotnet/coreclr/blob/6c12105bb8cc1821ba5d5c3d36aad609a44308e0/src/vm/methodtablebuilder.cpp#L1793

And then afterwards, `SetupMethodTable2` gets called

https://github.com/dotnet/coreclr/blob/6c12105bb8cc1821ba5d5c3d36aad609a44308e0/src/vm/methodtablebuilder.cpp#L1820

which is what sets intrinsic type:

https://github.com/dotnet/coreclr/blob/6c12105bb8cc1821ba5d5c3d36aad609a44308e0/src/vm/methodtablebuilder.cpp#L2040

It looks like @tannergooding just moved it in the last week:

https://github.com/dotnet/coreclr/commit/6b0164a9c94e8bc566914697a87db6b4267788e5

Should I move `SetIsIntrinsicType()` even earlier?  Outside of `MethodMethodTable2` and just before the call to `CheckIfSIMDAndUpdateSize()`?

Reply:
@eerhardt, it was actually moved even earlier, to just after the method table is created: https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L10444

Reply:
(method table is created here https://github.com/dotnet/coreclr/blob/master/src/vm/methodtablebuilder.cpp#L10130)

Reply:
It can be moved as early as needed. If necessary, it can be moved to even before MethodTable i s created. The flag can be stored in MethodTableBuilder first. A log of flags are on this plan already.

Reply:
`CheckIfSIMDAndUpdateSize()` is called before the MethodTable is created.  So I would need to store `isIntrinsicType` on MethodTableBuilder.  I'm not familiar at all with this code.  Do you have a suggestion on where I should put this flag?

Reply:
You can look at `fIsHardwareIntrinsic`

Reply:
Perfect - that's just what I needed.  Thanks